extensions/amp-carousel/0.1/slidescroll.js

```diff
@@ -287,6 +287,12 @@ export class AmpSlideScroll extends BaseSlides {
   }
 
   /** @override */
+  unlayoutCallback() {
+    this.slideIndex_ = null;
```

Comment:
I have now added a manual test (in `test/manual/amp-slidescroll.amp.html`) for the `amp-carousel` inside an `amp-lightbox`. The steps to reproduce are:

1. Click first thumbnail. Lightbox opens with first large image loaded.
2. Close lightbox.
3. Click third thumbnail. Lightbox opens with third large image loaded. Prior to the fix the third image would not load.

Before I added this test, I checked the `unlayoutCallback()` by putting a breakpoint on it. It did get called when the lightbox closed.